modules/@apostrophecms/schema/ui/apos/logic/AposInputArray.js

```diff
@@ -28,30 +28,81 @@ export default {
     const next = this.getNext();
     // this.schema is a computed property and is not available in data, that's why we use this.field.schema here instead
     const items = modelItems(next, this.field, this.field.schema);
-
     return {
       next,
       items,
+      isDragging: false,
       itemsConditionalFields: Object
-        .fromEntries(items.map(({ _id }) => [ _id, getConditionTypesObject() ]))
+        .fromEntries(items.map(({ _id }) => [ _id, getConditionTypesObject() ])),
+      emptyWhenIcon: this.field?.whenEmpty?.icon || 'text-box-multiple-icon',
+      emptyWhenLabel: this.field?.whenEmpty?.label || 'apostrophe:noItemsAdded'
     };
   },
   computed: {
+    isInlineTable() {
+      return this.field.style === 'table' && this.field.inline;
+    },
+    isInlineStandard() {
+      return this.field.style !== 'table' && this.field.inline;
+    },
+    isDraggable() {
+      if (this.field.draggable === false) {
+        return false;
+      }
+      if (this.field.readOnly) {
+        return false;
+      }
+      if (this.next.length > 1) {
+        return true;
+      }
+      return true;
+    },
+    isAddDisabled() {
+      return this.field.readOnly || (this.field.max && (this.items.length >= this.field.max));
+    },
+    inlineContextMenu() {
+      const menu = [
+        {
+          label: this.$t('apostrophe:remove'),
+          action: 'remove',
+          modifiers: [ 'danger' ]
+        }
+      ];
+      if (this.field.duplicate !== false) {
+        menu.unshift({
+          label: this.$t('apostrophe:duplicate'),
+          action: 'duplicate'
+        });
+      }
+      if (this.isDraggable) {
+        menu.unshift(
+          {
+            label: this.$t('apostrophe:moveUp'),
+            action: 'move-up'
+          },
+          {
+            label: this.$t('apostrophe:moveDown'),
+            action: 'move-down'
+          }
+        );
+      }
+      return menu;
+    },
```

Comment:
Optional: for a little less verbose version with a clear items order.
```suggestion
    inlineContextMenu() {
      return [
        ...this.isDraggable ? [
          {
            label: this.$t('apostrophe:moveUp'),
            action: 'move-up'
          },
          {
            label: this.$t('apostrophe:moveDown'),
            action: 'move-down'
          }
        ] : [],
        ...this.field.duplicated !== false ? [ {
          label: this.$t('apostrophe:duplicate'),
          action: 'duplicate'
        }] : [],
        {
          label: this.$t('apostrophe:remove'),
          action: 'remove',
          modifiers: [ 'danger' ]
        }
      ];
    },
```